src/SleighAsm.h

```diff
@@ -6,67 +6,274 @@
 #include <string>
 #include <vector>
 #include <r_core.h>
+#include <unordered_map>
 #include "architecture.hh"
 #include "sleigh_arch.hh"
+#include "SleighInstruction.h"
 
 class AsmLoadImage : public LoadImage
 {
-    private: 
-        RIO *io = nullptr;
-
-    public: 
-        AsmLoadImage(RIO *io) : LoadImage("radare2_program"), io(io) {}
-        virtual void loadFill(uint1 *ptr,int4 size,const Address &addr)
-        {
-            r_io_read_at(io, addr.getOffset(), ptr, size);
-        }
-        virtual string getArchType(void) const { return "radare2"; }
-        virtual void adjustVma(long adjust)
-        {
-            throw LowlevelError("Cannot adjust radare2 virtual memory");
-        }
+private:
+	RIO *io = nullptr;
+
+public:
+	AsmLoadImage(RIO *io): LoadImage("radare2_program"), io(io) {}
+	virtual void loadFill(uint1 *ptr, int4 size, const Address &addr)
+	{
+		r_io_read_at(io, addr.getOffset(), ptr, size);
+	}
+	virtual string getArchType(void) const { return "radare2"; }
+	virtual void adjustVma(long adjust)
+	{
+		throw LowlevelError("Cannot adjust radare2 virtual memory");
+	}
 };
 
 class AssemblySlg : public AssemblyEmit
 {
-	public:
-        char *str = nullptr;
+public:
+	char *str = nullptr;
+
+	void dump(const Address &addr, const string &mnem, const string &body) override
+	{
+		str = r_str_newf("%s %s", mnem.c_str(), body.c_str());
+	}
+
+	~AssemblySlg()
+	{
+		if(str)
+			free(str);
+	}
+};
+
+struct PcodeOperand
+{
+	PcodeOperand(): PcodeOperand(0x7fffffff, 0x7fffffff) {}
```

Comment:
Is this constructor used anywhere? Where do the `0x7fffffff` come from?

Reply:
This is constructor, used to specify it's uninitialized.
I will remove this.